src/pages/reservation/reserve/index.tsx

```diff
@@ -112,12 +131,18 @@ function ReservationPage() {
   const personnel = useSelector(
     (state: IPersonnelSliceState) => state.personnel.value,
   )
-
   const ticketPrice =
-    ticketDto && ticketDto.length > 0
-      ? ticketDto[0].price *
-        (personnel.adult + personnel.teenager + personnel.infant)
-      : 0 // 나중에 성인 청소년 소아 나눠지면 변경
+    ticketDto && personnel
+      ? ticketDto.reduce((acc: number, ticket: ITicketDto) => {
+          for (const personnelType in personnel) {
+            if (ticket.name === personnelType) {
+              acc += ticket.price * personnel[personnelType]
+              break
+            }
+          }
+          return acc
+        }, 0)
+      : 0
```

Comment:
P3: 가독성을 높이기 위해 
acc -> totalPrice
personnelType -> personnelCount
와 같이 변수명을 바꿔 사용하면 좋을 거 같아요. 

추가로 reduce와 for문 중첩을 피하고 ticket.name을 바로 'personnel' 객체에서 키로 사용하여 값을 가져온다면 코드 복잡성도 줄이고 성능 개선에도 도움이 될 것 같습니다.

```suggestion
ticketDto && personnel
  ? ticketDto.reduce((totalPrice, ticket) => {
      const personnelCount = personnel[ticket.name];
      if (personnelCount) {
        return totalPrice + (ticket.price * personnelCount);
      }
      return totalPrice;
    }, 0)
  : 0;
```